.github/workflows/ci.yml

```diff
@@ -34,27 +34,111 @@ jobs:
 
       - name: Compile tests
         id: compile-tests
-        # Note: profile changed because otherwise test binary is 1GB+ in size..
+        # Note:
+        # - profile changed because otherwise some test binaries are 100MB+ in size with pathfinder_lib over 1GB
+        # - package pathfinder produces two test binaries: pathfinder & pathfinder_lib
         run: |
-          cargo test --no-run --lib pathfinder -p pathfinder --locked --profile stripdebuginfo --message-format=json > /tmp/cargo-output.json
-          echo -n "TEST_EXECUTABLE=" >> $GITHUB_OUTPUT
-          jq -r 'select(.reason=="compiler-artifact" and .target.name=="pathfinder_lib") | .executable' /tmp/cargo-output.json >> $GITHUB_OUTPUT
+          cargo test --no-run --locked --profile stripdebuginfo --message-format=json \
+            -p pathfinder -p pathfinder-ethereum -p pathfinder-serde -p starknet-gateway-types \
+            | jq -r 'select(.reason=="compiler-artifact" and .target.test==true and .executable) | .executable' \
+            | xargs -I '{}' bash -c 'file={}; mv ${file} $(basename ${file::(-17)})'
 
-      # Finds the pathfinder_lib- test binary, and renames it and the .d file to something simpler to use.
-      # Essentially removing the hash to make running the file simpler (as it is used in other jobs).
-      - name: Rename test binary
-        run: mv ${{ steps.compile-tests.outputs.TEST_EXECUTABLE}} pathfinder_test
+      - name: Archive test binary pathfinder
+        uses: actions/upload-artifact@v3
+        with:
+          name: test-binary-pathfinder
+          path: pathfinder
+          if-no-files-found: error
```

Comment:
Why do we store this artifact at all? I mean the only test binary we're running in multiple steps is the one for `pathfinder-ethereum`, right? All the others could be just a simple `cargo test -p XXX` like we have for `pathfinder-common` and `pathfinder-retry`?

Or even better: isn't there a way to just run all tests _except_ the test for `pathfinder-ethereum` somehow? That way we wouldn't need to worry about not explicitly adding something to the CI again...

Reply:
> On your branch cargo test -p pathfinder --bin pathfinder still fails for me, so the fact that CI is green means that we're still missing some tests?

Hmm this is really weird - the test (`config::cli::tests::network`) does fail locally, but the action marks it as [pass](https://github.com/eqlabs/pathfinder/actions/runs/3601800297/jobs/6068051328):
```
test config::cli::tests::network ... ok
```

> Or even better: isn't there a way to just run all tests except the test for pathfinder-ethereum somehow? That way we wouldn't need to worry about not explicitly adding something to the CI again...

Sure, I think I can do that. The only downside is that tests for different test binaries will be executed in a sequence - right?

Reply:
I reworked it to run all tests excluding pathfinder-ethereum and then another 2 jobs that do pathfinder-ethereum on infura & alchemy.